src/lib/proof_system.ts

```diff
@@ -1,22 +1,40 @@
-import { Proof } from '../snarky';
+import { Bool, AsFieldElements } from '../snarky';
 
-export function proofSystem(target: any): any {}
+export { Proof, CompiledTag };
 
-export function branch(
-  target: any,
-  propertyName: string,
-  _descriptor?: PropertyDescriptor
-): any {}
+class Proof<T> {
+  static publicInputType: AsFieldElements<any> = undefined as any;
+  static tag: () => { name: string } = () => {
```

Comment:
the new `Proof` class needs a `tag` to identify the compiled `SmartContract` that this proof belongs to